src/common/reducers/repositories.js

```diff
@@ -2,10 +2,11 @@ import * as ActionTypes from '../actions/repositories';
 import { parseGitHubRepoUrl } from '../helpers/github-url';
 
 export function repositories(state = {
-  isFetching: false,
+  // has no server side content, so initial state is always 'fetching'
+  isFetching: true,
```

Comment:
Agree with @bartaz.

These three-state situations come up in components quite a lot, and whenever I've tried to reduce it to one binary state it hasn't worked. I think an "initial state", "needs data", "uninitialised" prop would be good, supporting the existing "isFetching" prop.